### ISSUE TITLE
S5: Switch AP and MTD to the numpy kernels; scikit-learn to dev deps; import-purity guard; close C-05 (#64 gate, part 2)

### ISSUE THREAD

**Issue by Polichinel** [views-platform/views-evaluation#64]
**Epic:** #66 · **Wave 2** · **Closes:** register **C-05** (Cluster C) · **Blocked by:** S4 (#70) — the parity suite must be merged with its red-on-injection evidence first · **Was:** the original #64, now split: S4 builds the gate, this story walks through it

## Background

`native_metric_calculators.py:4-7` imports `sklearn.metrics` at module level for `calculate_ap_native` (:79-84) and `calculate_mtd_native` (:124-130). scikit-learn imports pandas eagerly when it is installed, and `views_evaluation/__init__.py` imports the kernels, so **every module of this package puts pandas in `sys.modules`** as a side effect of being imported — measured per module in fresh subprocesses on 2026-09-16 (`numpy`, `scipy.stats`, `views_frames`: clean; `sklearn.metrics`: loads pandas). ADR-011's "zero knowledge of external data frameworks" and README:92's "independent of Pandas" are true of the source and false of the interpreter. Neither consumer imports or declares scikit-learn. After S4 the numpy kernels exist and have been proven equal.

## The work

1. Point `calculate_ap_native` and `calculate_mtd_native` at the numpy functions (return `float`); delete the sklearn import. Add the Red tests S4 could not: shape-guard and domain errors now raised by our code, through the public kernels.
2. `pyproject.toml`: remove `scikit-learn` from `[tool.poetry.dependencies]`; add `scikit-learn = "^1.6.0"  # Dev-only: test oracle for AP/MTD parity verification` to the dev group (the `properscoring` precedent, `:24-27`); reword the scipy comment at `:16` — scipy is now simply a direct dependency.
3. Import-purity guard (in `tests/test_metric_calculators.py` or `test_documentation_contracts.py`, not a new module): `assert importlib.util.find_spec("pandas") is not None` (fail, never skip — otherwise the guard is vacuous), then `subprocess.run([sys.executable, "-c", "import sys, views_evaluation; print('pandas' in sys.modules, 'sklearn' in sys.modules)"])` → `False False`. **Run it on the parent commit first and record it red.**
4. Runtime-only install job in `run_pytest.yml`: `poetry install --only main --all-extras && poetry run python -c "import views_evaluation, views_evaluation.evaluation.metric_frame"` — the only thing that will ever exercise "works without sklearn".
5. Docs: delete the Known Deviation at `documentation/ADRs/011_topology_and_dependency_rules.md:59` and `documentation/CICs/NativeEvaluator.md:140`; README `:262` → `NumPy + SciPy; no dataframe libraries`; `CHANGELOG.md` `[Unreleased]` `### Changed`: dependency change, no API change; the float64 and error-text deviations named.
6. Register: C-05 → Closed; Cluster C resolved; C-41 note (the resolve shrinks); C-07 closed row still true ("dev-only oracle").

**This story does not** touch `to_dataframe`, pandas, or the `dataframe` extra.

## Acceptance criteria

- [ ] `grep -rn sklearn views_evaluation/` returns nothing; `scikit-learn` appears in `pyproject.toml` only under the dev group
- [ ] Purity guard green with pandas installed, and **was red on the parent commit** (recorded)
- [ ] `test_ap_oracle_sklearn`, `test_mtd_known_tweedie` and the entire S4 parity suite pass **unmodified** (they now compare production numpy against the dev-only oracle — their permanent job)
- [ ] New Red tests for AP/MTD shape guard and domain errors through the public kernels
- [ ] Runtime-only install job green in CI
- [ ] ADR-011 and CICs/NativeEvaluator.md carry no sklearn deviation; README `:262` corrected
- [ ] Register: C-05 closed, Cluster C closed, header count decremented
- [ ] Full test suite still green; `ruff` clean; `validate_docs.sh` passes

## Validation

```bash
conda run --name views_pipeline pytest tests/ -q
conda run --name views_pipeline ruff check views_evaluation/ tests/
bash documentation/validate_docs.sh
conda run --name views_pipeline python -c "import sys, views_evaluation; print('pandas' in sys.modules, 'sklearn' in sys.modules)"   # False False
grep -rn sklearn views_evaluation/ && echo STILL PRESENT || echo gone

**Comment by Polichinel** [comment]
**Decision owed before this story switches dispatch (found by `/code-review max` on S4/#70).**

`_average_precision_numpy` transcribes scikit-learn's all-negative-truth behaviour: return `0.0` and emit a plain `UserWarning` ("No positive class found in y_true, recall is set to one for all thresholds."). That is what the public kernel does **today** through sklearn, so S4 reproduces it for parity. But it is a non-raising return with no ADR-015 ruling, and once dispatched it becomes a first-party warning on every classification group whose truth is all zeros — routine in conflict data — with the uncontracted `0.0` averaged into `to_metric_frame()`'s `"mean"` row (Pearson's `nan` sentinel, by contrast, is excluded by `nanmean`).

Options, per ADR-015's fault-vs-data-property test:
- **A — contracted sentinel, warning suppressed** (the `calculate_pearson_native` precedent): return `nan` ("no positive class: AP undefined for this group"), suppress the warning, document in ADR-015's rulings table and CICs/MetricCatalog.md. Changes the mean row (group excluded rather than pulled toward 0).
- **B — raise**: an all-negative group is a configuration/data problem the caller should see. Aborts the evaluation, as the reverted Pearson raise did on 2026-08-02.
- **C — keep `0.0`, suppress the warning, document** as a ruled sentinel. Preserves today's numbers exactly.

Whichever is chosen, the S4 parity test `test_all_negative_truth_warns_and_is_zero` is then rewritten to assert the ruling rather than sklearn parity, and ADR-015 gains the row. Recorded here so the switch cannot happen without it.

**Comment by Polichinel** [comment]
**Decided 2026-09-18 by the maintainer: Option A — `nan` sentinel, warning suppressed.**

Recorded as ADR-015 ruling 9 (R9, with the full weighing of A/B/C), register decision **D-01** (a new *Decisions* section), and `CICs/MetricCatalog.md`'s degenerate-input table — all landing in this story's PR alongside the switch. In brief: an empty group is a property of conflict data (the R1/R2 case), so it is ruled the way MCR and Pearson are; raising aborts every metric for a routine condition (tried on Pearson, reversed same day); keeping `0.0` codifies a wrong number that silently drags the `mean` row toward zero.

Consequence to carry into S6's release notes (ADR-022 §3): `views-models/ensembles/rusty_bucket` (the one live config requesting `AP`) and views-reporting's canonical classification cell see `nan` for an empty group and a `mean` over the rest, where before they saw `0.0` and a lower mean.

**Comment by Polichinel** [comment]
Merged to `development` in #78. Ritual: /review-diff, /code-review max (15 findings), /falsify guard (24 mutations, 10 survivors closed), /register-risk, /ship-it. 968 tests green. C-05 and Cluster C closed; ADR-015 R9 / register D-01 carry the maintainer's ruling and its consequences (pipeline-core's `AP_mean` under `nanmean`-less averaging; all-sentinel frame warns). The CI run of this PR was the first execution of the runtime-only install step.
